rviz_common/src/rviz_common/frame_manager.cpp

```diff
@@ -86,14 +89,15 @@ void FrameManager::update()
   if (!pause_) {
     switch (sync_mode_) {
       case SyncOff:
-        sync_time_ = rclcpp::Time::now();
+        sync_time_ = clock_->now();
         break;
       case SyncExact:
         break;
       case SyncApprox:
         // adjust current time offset to sync source
-        current_delta_ = 0.7 * current_delta_ + 0.3 * sync_delta_;
-        sync_time_ = rclcpp::Time(rclcpp::Time::now().nanoseconds() - current_delta_);
+        current_delta_ = static_cast<uint64_t>(0.7 * current_delta_ + 0.3 * sync_delta_);
```

Comment:
Is the static cast required? It is implicitly being converted to `uint64_t` on assignment.

Reply:
I added it because my IDE showed a warning.
But I agree, it is implicitly converted and therefore not required.

Reply:
While the static cast is not required I preferred it as it expresses that the otherwise implicit conversion from a double to a integer is intentional.